libs/qtandroidserialport/src/qserialport_android.cpp

```diff
@@ -109,57 +118,75 @@ QSerialPortPrivate::QSerialPortPrivate(QSerialPort *q)
 bool QSerialPortPrivate::open(QIODevice::OpenMode mode)
 {
     rwMode = mode;
-    __android_log_print(ANDROID_LOG_INFO, V_TAG, "Opening %s", systemLocation.toLatin1().data());
+    __android_log_print(ANDROID_LOG_INFO, kJTag, "Opening %s", systemLocation.toLatin1().data());
+
+    __android_log_print(ANDROID_LOG_INFO, kJTag, "Calling Java Open");
+    QAndroidJniObject jnameL = QAndroidJniObject::fromString(systemLocation);
+    cleanJavaException();
+    deviceId = QAndroidJniObject::callStaticMethod<jint>(
+        kJniClassName,
+        "open",
+        "(Ljava/lang/String;I)I",
+        jnameL.object<jstring>(),
+        (jint)this);
+    cleanJavaException();
+
+    isReadStopped = false;
+
+    if (deviceId == BAD_PORT)
+    {
+        __android_log_print(ANDROID_LOG_ERROR, kJTag, "Error opening %s", systemLocation.toLatin1().data());
+        q_ptr->setError(QSerialPort::DeviceNotFoundError);
+        return false;
+    }
 
     if (!hasRegisteredFunctions)
     {
+        __android_log_print(ANDROID_LOG_INFO, kJTag, "Registering Native Functions");
         //  REGISTER THE C++ FUNCTION WITH JNI
-        QAndroidJniEnvironment envL;
-
-        JNINativeMethod methodsL[] {
+        JNINativeMethod javaMethods[] {
             {"nativeDeviceHasDisconnected", "(I)V",                   reinterpret_cast<void *>(jniDeviceHasDisconnected)},
             {"nativeDeviceNewData",         "(I[B)V",                 reinterpret_cast<void *>(jniDeviceNewData)},
             {"nativeDeviceException",       "(ILjava/lang/String;)V", reinterpret_cast<void *>(jniDeviceException)}
         };
 
-        QAndroidJniObject javaClassL(V_jniClassName);
-        jclass objectClassL = envL->GetObjectClass(javaClassL.object<jobject>());
-        jint valL = envL->RegisterNatives(objectClassL, methodsL, sizeof(methodsL) / sizeof(JNINativeMethod));
-        envL->DeleteLocalRef(objectClassL);
+        QAndroidJniEnvironment jniEnv;
+        if (jniEnv->ExceptionCheck()) {
+            jniEnv->ExceptionDescribe();
+            jniEnv->ExceptionClear();
+        }
+
+        QAndroidJniObject javaClass(kJniClassName);
+        if(!javaClass.isValid()) {
+            __android_log_print(ANDROID_LOG_ERROR, kJTag, "Java class %s not valid", kJniClassName);
```

Comment:
This is the main Issue (for now) with the SerialPort glue code. Here is where we register the c++ function for the Java side to call. For whatever reason, this part of the code is not finding the * kJniClassName* class (and I'm not allowing the code to proceed). This same class name is used for all outgoing calls (c++ to Java) and working just fine.